peer.go

```diff
@@ -63,6 +64,64 @@ func (id ID) MatchesPublicKey(pk ic.PubKey) bool {
 	return oid == id
 }
 
+func (id ID) ExtractEd25519PublicKey() ic.PubKey {
+	// ed25519 pubkey identity format
+	// <identity mc><length (2 + 32 = 34)><ed25519-pub mc><ed25519 pubkey>
+	// <0x00       ><0x22                ><0xed01        ><ed25519 pubkey>
```

Comment:
You can do binary.PutUvarint with the value. It is compliant with our varint spec up until 63 bits of to be encoded value.